pkgs/build-support/haxe/override_haxe_lib_from_hmm.nix

```
@@ -0,0 +1,103 @@
+# This tool can be used to convert as set of derivation containing haxe dependncies (as from haxe-packages.nix) to use version specified in an hmm.json file, used by certain project to lock inputs.
+# To update (or initiate it) for a package:
+# 1. copy the hmm.json file in the package folder
+# 2. run update_hmm_hashes.py (you will to have nix-prefetch-git in the PATH, which is packaged in nixpkgs under the same name).
+#    It take as argument a path to the hmm.json file, and a path to an output json file that contains the hashes for the dependancies (usually named hashes.json)
+{
+  lib,
+  haxePackages,
+  fetchgit,
+  fetchzip,
+}:
+
+{
+  hmm_attrset ? null,
+  hmm_file ? null,
+  hashes_attrset ? null,
+  hashes_file ? null,
+  packages,
+}:
+
+assert lib.xor (hmm_attrset != null) (hmm_file != null);
+assert lib.xor (hashes_attrset != null) (hashes_file != null);
+
+let
+  hmm = if hmm_attrset != null then hmm_attrset else (builtins.fromJSON (builtins.readFile hmm_file));
+  hashes =
+    if hashes_attrset != null then
+      hashes_attrset
+    else
+      (builtins.fromJSON (builtins.readFile hashes_file));
+
+  index_override = builtins.listToAttrs (
```

```suggestion
  index_override = lib.listToAttrs (
```

In general, in nixpkgs, we prefer to import from `lib` over builtins. That enables us to support the widest array of Nix versions (including ones in which a specific builtin doesn't exist.)

In this case, it doesn't matter: listToAttrs is very longstanding.